org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -192,6 +213,8 @@ public void cleanupResources(IProject project) throws CoreException {
 					File f = new File(cu.getUnderlyingResource().getLocationURI());
 					if (!f.exists()) {
 						cu.delete(true, null);
+					} else {
+						JDTUtils.resolveCompilationUnit(f.toURI());
```

Comment:
@snjeza Do you remember why this line is needed?

Reply:
You can check the issue by adding che (https://github.com/eclipse/che) to a workspace.